### ISSUE TITLE
SIGSEV

### ISSUE THREAD

**Issue by easonwang717** [plasma-umass/scalene#556]
python 3.8.16
pytorch 1.13.1
scalene 1.5.19

Report SIGSEGV

To be more specific, I got profiling results for several python module files which are called by the main. I can only get profiling results for one python file which consumes the most execution time.

**Comment by emeryberger** [comment]
This is not really enough information for us to meaningfully try to resolve the issue. Please see if you can come up with a case that reproduces it. Thanks.

**Comment by easonwang717** [comment]
> This is not really enough information for us to meaningfully try to resolve the issue. Please see if you can come up with a case that reproduces it. Thanks.

Yes. I want to profile my previous project's code.

1. git clone https://github.com/easonwang717/FedCor_prof.git
2. install PyTorch and cvxopt (conda install -c Conda-forge cvxopt)
3. run a toy case with the script below:
scalene --json --outfile fedcor.json src/federated_main.py --GPU=2 --dataset=cifar --model=cnn --kernel_sizes 3 3 3 --num_filters 32 64 64 --mlp_layer 64 --epochs=10 --num_user=100 --frac=0.05 --local_ep=5 --local_bs=50 --lr=0.01 --lr_decay=1.0 --optimizer=sgd --reg=3e-4 --iid=0 --unequal=0 --alpha=0.2 --verbose=1 --seed 1 --gpr --gpr_selection --discount_method=time --discount=0.9 --GPR_interval=5 --group_size=51 --GPR_gamma=0.97 --poly_norm=0 --warmup=2

Then I finally got SIGSEV.

Thanks!

**Comment by sternj** [comment]
Failed to replicate on Python 3.8.16 and `miniconda3-3.8-4.12.0`. Does the SEGV occur at the end of the program?

**Comment by easonwang717** [comment]
> 

Yes, it does occur SEGV at the end when I scalene... What specific failure did you see?

**Comment by easonwang717** [comment]
> Failed to replicate on Python 3.8.16 and `miniconda3-3.8-4.12.0`. Does the SEGV occur at the end of the program?

I don't know miniconda.... I use Conda...

**Comment by sternj** [comment]
Does Scalene output results before the segv occurs?

**Comment by easonwang717** [comment]
> Does Scalene output results before the segv occurs?

Yes, I have partial results.

**Comment by easonwang717** [comment]
In addition, when I just want to profile the cpu using:

scalene  --cpu --json --outfile fedcor_cpu.json src/federated_main.py --GPU=2 --dataset=cifar --model=cnn --kernel_sizes 3 3 3 --num_filters 32 64 64 --mlp_layer 64 --epochs=10 --num_user=100 --frac=0.05 --local_ep=5 --local_bs=50 --lr=0.01 --lr_decay=1.0 --optimizer=sgd --reg=3e-4 --iid=0 --unequal=0 --alpha=0.2 --verbose=1 --seed 1 --gpr --gpr_selection --discount_method=time --discount=0.9 --GPR_interval=5 --group_size=51 --GPR_gamma=0.97 --poly_norm=0 --warmup=2

I got the error: unrecognized arguments: src/federated_main.py

Is there any bugs in scalene when parsing arguments into the python file?

**Comment by sternj** [comment]
Please put that in another issue report. I believe that the primary issue you're running into is a known one that we are currently working on, but have been unable to reproduce in our own testbeds. We will post a fix in the coming days and notify you on the PR so you can test.

**Comment by emeryberger** [comment]
Also, Scalene usage, from `--help` (you should terminate all Scalene arguments with `---`).

```Python
command-line:
  % scalene [options] your_program.py [--- --your_program_args] 
or
  % python3 -m scalene [options] your_program.py [--- --your_program_args] 
```
